backend/src/test/java/com/rhyun/backend/security/service/UserServiceTest.java

```diff
@@ -119,4 +120,72 @@ void getLoggedInUserTest_whenLoggedInUserDoesNotExist_thenThrow() {
         verify(authentication, times(1)).getPrincipal();
         verify(securityContext, times(1)).getAuthentication();
     }
+
+    @Test
+    void findUserByIdTest_whenIdExists_thenReturnUser() {
+        // GIVEN
+        AppUser user = new AppUser("1", "user1", "password1", AppUserRole.ADMIN);
+        when(userRepository.findById("1")).thenReturn(Optional.of(user));
+
+        // WHEN
+        UserDto actual = userService.findUserById("1");
+
+        // THEN
+        UserDto expected = new UserDto(user.username(), user.password(), user.role());
+        assertEquals(expected, actual);
+        verify(userRepository, times(1)).findById("1");
+    }
+
+    @Test
+    void findUserByIdTest_whenIdDoesNotExist_thenThrow() {
+        // GIVEN
+        when(userRepository.findById("1")).thenReturn(Optional.empty());
+
+        // WHEN
+        // THEN
+        assertThrows(
+                UserNotFoundException.class,
+                () -> userService.findUserById("1")
+        );
+        verify(userRepository, times(1)).findById("1");
+
+        assertThrows(UsernameNotFoundException.class, () -> userService.findUserByUsername("user1"));
+        verify(userRepository, times(1)).findUserByUsername("user1");
+    }
+
+    @Test
+    void updateUserTest_whenIdExists_thenUpdateUserEntity() {
+        // GIVEN
+        AppUser originalUser = new AppUser("1", "user1", "password1", AppUserRole.ADMIN);
+        PutUserDto putUserDto = new PutUserDto("pass123", AppUserRole.EMPLOYEE);
+        AppUser updatedUser = new AppUser("1", "user1", passwordEncoder.encode(putUserDto.password()), putUserDto.role());
```

Comment:
since you mock passwordEncoder, would you not also require a `when(passwordEncoder.encode("password1").thenReturn("encodedPassword")` for example? And then use that password in updatedUser
```suggestion
        AppUser updatedUser = new AppUser("1", "user1", "encodedPassword", putUserDto.role());
```

Reply:
same below in the update test when Id does not exist, but you could also verify that passwordEncoder.encode() is never called and not specify the `when(passwordEncoder...`